server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -137,16 +166,20 @@ public void getSynonymsSets(int from, int size, ActionListener<PagedResult<Synon
         client.prepareSearch(SYNONYMS_ALIAS_NAME)
             .setSize(0)
             .addAggregation(
-                new TermsAggregationBuilder(SYNONYM_SETS_AGG_NAME).field(SYNONYMS_SET_FIELD)
-                    .order(BucketOrder.key(true))
-                    .size(MAX_SYNONYMS_SETS)
+                new FilterAggregationBuilder(RULESET_FILTER_AGG_NAME, QueryBuilders.termQuery(OBJECT_TYPE_FIELD, SYNONYM_RULE_OBJECT_TYPE))
```

Comment:
We need to filter out the synonym set documents.

We could also cheat, and substract one from the result obtained (as there will always be a single doc that represents the synonym set). But somehow this makes my conscience more at ease.

Reply:
Feels like that would be a good place to include the "why" in a comment

Reply:
Instead of `filter` aggregation, we should use a query that filters only only synonym rules objects, as this will be [faster](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-filter-aggregation.html#use-top-level-query-to-limit-all-aggs).

Reply:
Thanks Mayya! Implementing that 👍